Odometry.cpp

```diff
@@ -41,50 +42,76 @@ void Odometry::update() {
+ */
 double Odometry::getDistanceTraveled() {
     return distance;
 }
 
+/* 
+ * Returns current estimate for the velocity of the robot from the wheels
+ */ 
+double Odometry::getVelocity() {
+    return Odometry::getVelocity(NULL);
+}
+
+/*
+ * Returns current estimate for velocity of the robot from the wheels
+ * Use this method if you need individual wheel estimates
+ * Velocity for Wheel A is in index 0 and Wheel B is in index 1
+ */
 double Odometry::getVelocity(double* wheel_velocities) {
     double average_a = velocity_buffer_a_.average();
     double average_b = velocity_buffer_b_.average();
     if (wheel_velocities) {
+        // put individual wheel velocity averages into array
         wheel_velocities[0] = average_a;
         wheel_velocities[1] = average_b;
     }
     return (average_a + average_b)/2;
 }
 
 double Odometry::getHeading() {
-    return theta;
+    return theta; 
 }
 
 double Odometry::getHeadingDegrees() {
     return theta * RADS_DEGREE;
 }
 
+/*
+ * Calculates the distance traveled by each wheel
+ * Note currently at this point a reverse rotation of the wheels decreaases the wheel
+ */ 
 void Odometry::calculateDistanceTotal() {
      // Converting raw counts into revolutions
-    double total_revolutions_a = ((double) last_motor_count_a)/ (double) COUNTS_PER_REVOLUTION;
-    double total_revolutions_b = ((double) last_motor_count_b)/ (double) COUNTS_PER_REVOLUTION;
+    double total_revolutions_a = (double) last_motor_count_a/ (double) COUNTS_PER_REVOLUTION;
+    double total_revolutions_b = (double) last_motor_count_b/ (double) COUNTS_PER_REVOLUTION;
     double distance_a = total_revolutions_a * WHEEL_CIRCUMFERENCE;
     double distance_b = total_revolutions_b * WHEEL_CIRCUMFERENCE;
     distance = (distance_a + distance_b) / 2;
 }
-
+/*
+ * Calculates the instananeous linear velocity for each wheel
+ */
 void Odometry::calculateVelocityInstantaneous(long delta_time, long delta_a, long delta_b) {
-    angular_velocity_a = ((double) delta_a)/delta_time;
-    angular_velocity_b = ((double) delta_b)/delta_time;
+    if (!delta_time) { return; } // Prevents divide by zero error
+    angular_velocity_a = (double) delta_a/(double) delta_time;
+    angular_velocity_b = (double) delta_b/(double) delta_time;
     double current_velocity_a = angular_velocity_a * WHEEL_RADIUS;
     double current_velocity_b = angular_velocity_b * WHEEL_RADIUS;
     velocity_buffer_a_.push(current_velocity_a);
     velocity_buffer_b_.push(current_velocity_b);
 }
 
+/*
+ * Derives a heading in radians from distance traveled by each wheel
+ */
 void Odometry::calculateHeadingInstantaneous(long delta_a, long delta_b) {
-    // Converting raw counts into revolutions
-    double revolutions_a = ((double) delta_a)/(double) COUNTS_PER_REVOLUTION;
-    double revolutions_b = ((double) delta_b)/(double) COUNTS_PER_REVOLUTION;
+    if (!delta_a && !delta_b) { return; } // Early return if there are no changes
```

Comment:
Don't you want an OR hear? Assuming `delta_a` is 0 but `delta_b` is 1, you will have not(0) && not(1) which is 1 && 0 which is false. But you need to check to see if at minimum one of those 2 is 0, correct?

Reply:
This is there purely to handle the case that at the beginning there is no movement at all or it has stop moving.